platform/qt/include/qmapboxgl.hpp

```diff
@@ -232,15 +225,29 @@ public slots:
 
 signals:
     void needsRendering();
-    void mapChanged(QMapboxGL::MapChange);
     void copyrightsChanged(const QString &copyrightsHtml);
 
+signals:
+    void cameraWillChange(QMapboxGL::CameraChangeMode);
```

Comment:
Rationale is that instead of having a switch case to catch the specific notifications from `cameraWillChange`, you connect to the specific signals you need.

Also, the current implementation does not provide the extra arguments passed e.g. via [onDidFailLoadingMap](https://github.com/mapbox/mapbox-gl-native/blob/master/include/mbgl/map/map_observer.hpp#L33) or [onSourceChanged](https://github.com/mapbox/mapbox-gl-native/blob/master/include/mbgl/map/map_observer.hpp#L39).